src/core/surface/call.c

```diff
@@ -205,8 +205,8 @@ struct grpc_call {
   /* Received call statuses from various sources */
   received_status status[STATUS_SOURCE_COUNT];
 
-  void *context[GRPC_CONTEXT_COUNT];
-  void (*destroy_context[GRPC_CONTEXT_COUNT])(void *);
+  /* Contexts for various subsystems (security, tracing, ...). */
+  grpc_call_context contexts[GRPC_CONTEXT_COUNT];
```

Comment:
I'm starting to think that it might be worth holding these as separate arrays here... and providing just the first in transport_op (so no access to destroy_context from there).

It'll mean we need to do a copy in _init, but it's a tiny copy, and makes this interface a little safer/easier to reason about.

What do you think though? You've used this more :)